### PR TITLE
Remoting implementation over Akka

### PR DESCRIPTION
**Pull request by lalithsuresh**
This PR includes the first go at the remoting implementation over Akka. The unsubscribe/cancellation part is still WIP. The code also needs testing (and subsequent hardening) for cases where the messages don't make it across the wire etc. 

The PR also includes a fix for an NPE in the toString() method of PartialResult.

### CODE REVIEW

File: src/main/java/org/hiero/sketch/remoting/SketchClientActor.java [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.atomic.AtomicInteger;
+
+/**
+ * This actor wires a local observable pertaining to a map or sketch
+ * operation with its remote execution. Submitting a {Map,Sketch}Operation
+ * to this actor returns an observable that streams values in accordance
+ * to its execution on the remote node.
+ */
+public class SketchClientActor extends AbstractActor {
```
Comment (mbudiu-vmw) [inline src/main/java/org/hiero/sketch/remoting/SketchClientActor.java]: We'll also have to figure out how to GC these guys. Do you know how the system keeps track of actors?
We'd like for actors without references to be GC-ed. At the same time they should also initiate the GC of the remote counterparts.

File: src/main/java/org/hiero/sketch/dataset/RemoteDataSet.java [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+import scala.concurrent.duration.Duration;
+
+import static akka.pattern.Patterns.ask;
+
+/**
+ * An IDataSet that is a proxy for a DataSet on a remote machine.
+ */
+public class RemoteDataSet<T> implements IDataSet<T> {
+    private final ActorRef clientActor;
+
```
Comment (lalithsuresh) [inline src/main/java/org/hiero/sketch/dataset/RemoteDataSet.java]: Will fix.

File: src/main/java/org/hiero/sketch/dataset/RemoteDataSet.java [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+import akka.util.Timeout;
+import org.hiero.sketch.dataset.api.*;
+import org.hiero.sketch.remoting.MapOperation;
+import org.hiero.sketch.remoting.SketchOperation;
+import rx.Observable;
+import scala.concurrent.Await;
+import scala.concurrent.Future;
+import scala.concurrent.duration.Duration;
+
+import static akka.pattern.Patterns.ask;
```
Comment (lalithsuresh) [inline src/main/java/org/hiero/sketch/dataset/RemoteDataSet.java]: Wow, much derp. Will fix.

File: src/main/java/org/hiero/sketch/dataset/RemoteDataSet.java [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+            obs = (Observable<PartialResult<R>>) Await.result(future, timeout.duration());
+        } catch (final Exception e) {
+            e.printStackTrace();
+        }
+        return obs;
+    }
+
+    @Override
+    public <S> Observable<PartialResult<IDataSet<Pair<T, S>>>> zip(final IDataSet<S> other) {
+        return null;
```
Comment (lalithsuresh) [inline src/main/java/org/hiero/sketch/dataset/RemoteDataSet.java]: Forgot about this method. Will do.

File: src/main/java/org/hiero/sketch/remoting/SketchServerActor.java [hunk 5]
```diff
@@ -0,0 +1,178 @@
...
+ */
+public class SketchServerActor<T> extends AbstractActor {
+
+    private final IDataSet<T> dataSet;
+    private final ConcurrentHashMap<UUID, Subscription> operationToObservable
+            = new ConcurrentHashMap<>();
+
+    private static final AtomicInteger nodeId = new AtomicInteger(0);
+    private static final Config config = ConfigFactory.parseString(
+        "akka {\n" +
```
Comment (lalithsuresh) [inline src/main/java/org/hiero/sketch/remoting/SketchServerActor.java]: Yes, I'm on it for some clustering support I'm adding.

File: src/main/java/org/hiero/sketch/remoting/SketchClientActor.java (hunk 1 above)
Comment (lalithsuresh) [inline src/main/java/org/hiero/sketch/remoting/SketchClientActor.java]: Working on that for now in the form of a book-keeping server.

**Pull request merged**